doc/tb3_tutorial/README.md

```diff
@@ -0,0 +1,573 @@
+
+![enter image description here](https://github.com/mlherd/ros2/blob/master/img/Rviz_start.PNG?raw=true)
+
+Now, the map should appear in Rviz. Next we need to set the estimate 2D Position of our Turtlebot on the map. Looking at the simulator in Gazebo we know that our Turtlebot is located at the bottom right in the map.
+
+![enter image description here](https://github.com/mlherd/ros2/blob/master/img/Rviz_initial.PNG?raw=true)
+
+Let's click on the 2D Pose Estimate button and point the location of our robot on the map. The direction of the green arrow is the orientation of Turtlebot.
+
+    -  Localize the robot using “2D Pose Estimate” button.
+    -  Make sure all transforms from odom are present. (odom->base_link->base_scan)
+
+First, find where the robot is in Gazebo. You can see where the robot's initial position in Gazebo.
+
+![enter image description here](https://github.com/mlherd/ros2/blob/master/img/gazebo_turlebot1.PNG?raw=true)
+Then set the pose of the robot in Rviz.
+
+![enter image description here](https://github.com/mlherd/ros2/blob/master/img/Rviz2D_Pose_Estimate.PNG?raw=true)
+Now, the 3D model of Turtlebot should move to that location. A small error in estimated location is OK.
+
+Next, we will pick a target location for Turtlebot in the map.  We can send our robot a goal position and a goal orientation by using the **Navigation2 Goal** button.
+
+*Note*: this uses a ROS2 Action to send the goal, and a pop-up window will appear on your screen with a 'cancel' button if you wish to cancel.
+
+![enter image description here](https://github.com/mlherd/ros2/blob/master/img/RvizNavigation2Goal1.PNG?raw=true)
+
+Once we define the target location, Turtlebot will create a global path and start navigating in the map.
+
+![enter image description here](https://github.com/mlherd/ros2/blob/master/img/RvizNavigation2Goal2.PNG?raw=true)
+You can also observe that our Turtlebot move in the simulated environment in Gazebo as well.
+
+![enter image description here](https://github.com/mlherd/ros2/blob/master/img/gazebo_turtlebot3.PNG?raw=true)
+This steps concludes the first part of this tutorial. In the fallowing sections, we are going to learn how to control the real Turtlebot.
+
+![enter image description here](https://raw.githubusercontent.com/mlherd/ros2/master/img/nav2_gif.gif)
+
+## Testing Navigation 2 with a real Turtlebot 3
+- This step soon to be completed. We are waiting for Turtlebot 3 firmware release for Dashing.
+###  Setting Up Turtlebot Waffle
+- You can follow the instruction [here](http://emanual.robotis.com/docs/en/platform/turtlebot3/ros2/).  Complete following steps on Turtlebot3.
+	- 1.2
+	- 1.3
+	- 2.1
+####  Test Navigation2 with the real Turtlebot Waffle**
+Coming Soon!
+
+## More About Navigation 2
+
+![enter image description here](https://raw.githubusercontent.com/mlherd/ros2/master/img/uml_diagram.jpg)
```

Comment:
Spelling mistake in diagram. Gazibo should be Gazebo.

Reply:
Uploaded a new uml diagram.